### PR TITLE
Remove mandatory domain check in the self Registration Autologin Cookie

### PR DESCRIPTION
**Pull request by ShehanDinuka**
## Purpose
Fix self registration auto login IsEnable check [Fix: #10603](https://github.com/wso2/product-is/issues/10603)

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 1]
```diff
@@ -116,7 +117,8 @@ public AuthenticatorFlowStatus process(HttpServletRequest request,
 
         if (context.isLogoutRequest()) {
             return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
-        } else if (autoLoginCookie != null && isEnableAutoLoginAfterPasswordReset(context)) {
+        } else if (autoLoginCookie != null &&
```
Comment (gayashanbc) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:120]: Refer https://github.com/wso2/identity-apps/pull/1409#discussion_r536169173

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 2]
```diff
@@ -691,6 +693,17 @@ private boolean isEnableAutoLoginAfterPasswordReset(AuthenticationContext contex
...
     }
 
+    boolean isEnableSelfRegistrationAutoLogin(AuthenticationContext context)
+            throws AuthenticationFailedException {
+
+        try {
+            return Boolean.parseBoolean(Utils.getConnectorConfig(SELF_REGISTRATION_AUTO_LOGIN,
+                            context.getTenantDomain()));
+        } catch (IdentityEventException e) {
+            throw new AuthenticationFailedException("Error occurred while resolving isEnableAutoLogin property.", e);
```
Comment (gayashanbc) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:703]: Specify which auto login property in the description

**Review by gayashanbc — COMMENTED** [review]
Added comments

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java [hunk 3]
```diff
@@ -86,16 +88,9 @@
...
-import static org.powermock.api.mockito.PowerMockito.doAnswer;
-import static org.powermock.api.mockito.PowerMockito.mock;
-import static org.powermock.api.mockito.PowerMockito.mockStatic;
-import static org.powermock.api.mockito.PowerMockito.when;
-import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.assertNotNull;
-import static org.testng.Assert.assertTrue;
+import static org.mockito.Matchers.*;
+import static org.powermock.api.mockito.PowerMockito.*;
+import static org.testng.Assert.*;
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java:93]: Shall we remove the wildcard imports?

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 4]
```diff
@@ -171,10 +171,6 @@ protected AuthenticatorFlowStatus executeAutoLoginFlow(HttpServletRequest reques
         String flowType = resolveAutoLoginFlow(autoLoginCookie.getValue());
         if (AutoLoginConstant.SIGNUP.equals(flowType)) {
             alias = getSelfRegistrationAutoLoginAlias(context);
-            String domainInCookie = (String) contentJSON.get(AutoLoginConstant.DOMAIN);
-            if (StringUtils.isEmpty(domainInCookie)) {
-                throw new AuthenticationFailedException("The domain name is not available in the ALOR cookie content.");
-            }
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:177]: What was the reason to remove this logic?

**Comment by ashensw** [comment]
@ShehanDinuka seems the PR title does not match with the fix? Can you check and change it?

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java (hunk 4 above)
Comment (ShehanDinuka) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:177]: Domain in the autologin cookie is not used anywhere and it should not be a mandatory check

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java (hunk 3 above)
Comment (ShehanDinuka) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java]: Done

**Review by ashensw — APPROVED** [review]

**Pull request merged**